### DISCUSSION TITLE
Allow passing mangle options to swc minify, such as keepClassnames and keepFnames

### DISCUSSION THREAD

**Discussion by JCMais** [vercel/next.js#39160] [Ideas]
### Describe the feature you'd like to request

Allow passing the [mangle options](https://swc.rs/docs/configuration/minification) to swc minify

### Describe the solution you'd like

I would like to be able to set the following mangle options when using the swc minify:
```
  mangle: {
      keepClassnames: true,
      keepFnames: true
  }
```

This is important when using class constructor names in the code itself.

### Describe alternatives you've considered

The only alternative is using a patch to add those directly to this file: https://github.com/vercel/next.js/blob/d091c818c5a66c7137275e2b0b711bc105898e19/packages/next/build/webpack/plugins/terser-webpack-plugin/src/index.js#L169

**Comment by balazsorban44** [comment]
Although it sounds like you need this for a different reason, this is similar to #37242. You might want to leave a comment there. :+1:

**Reply by JCMais** [reply]
thanks @balazsorban44 , we can close this one.

**Comment by ivadenis** [comment]
I've just spent a big chunk of my life figuring out why 3rd party web worker breaks with production build. Turns out that the mangler would obfuscate WorkerGlobalScope property `self` which caused a Reference Error. After upgrading `next` to the latest version and using
```
 swcMinifyDebugOptions: {
        mangle: {
          reserved: ['self', 'global']
        }
      }...
```
the issue has been fixed.

While doing a build, I noticed this warning `warn  - SWC minify debug option specified. This option is for debugging minifier issues and will be removed once SWC minifier is stable.` 

I just want to note the importance of being able to set the minifier/mangler settings from within a config.

**Reply by connorads** [reply]
Hey @ivadenis, now that `swcMinifyDebugOptions` has been removed, have you got a workaround for mangling issues?

**Reply by ivadenis** [reply]
> Hey @ivadenis, now that `swcMinifyDebugOptions` has been removed, have you got a workaround for mangling issues?

Im still on next 12.

**Comment by billnbell2** [comment]
we need it added back~~~

**Comment by logusgraphics** [comment]
I was facing a similar issue, it seems [they added a flag](https://github.com/vercel/next.js/pull/42633) to the `next build --no-mangling` command.  This can be verified by the log:

```Mangling is disabled. Note: This may affect performance and should only be used for debugging purposes```

But I was still facing issues of function and class names being mangled. So I had to disable the minification in `next.config.js`:

```javascript
const nextConfig = {
  swcMinify: false,
  ...
};
```
And that did it for me. However there is a warning:
```
Disabling SWC Minifer will not be an option in the next major version. Please report any issues you may be experiencing to https://github.com/vercel/next.js/issues
```

**Comment by billnbell2** [comment]
We are using Joyride https://github.com/gilbarbara/react-joyride and it does not work without setting `swcMinify: false,`...

Unsure why it mangles things where it does not work? Any ideas?
